### PR TITLE
Add new filter rule:  DegreesOfSeperation

### PR DESCRIPTION
**Pull request by Mattkmmr**
The filter rule allows to filter for relatives by their degrees of seperation to the selected person. The code uses a modified version of the IsLessThanNthGenerationAncestorOf filter rule for ancestor and descandants calculation. It is also possible to include the partners of the descandants.

### CODE REVIEW

File: FilterRules/degreesofseparation.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+        get_partners = bool(int(self.list[2]))
+        if get_partners:
+            self.__get_partners()
+
+    def __get_root_handle(self):
+        """Get the handle of the starting person."""
+        pid = self.list[0]
+        person = self.db.get_person_from_gramps_id(pid)
+        root_handle = person.get_handle()
```
Comment (prculley) [inline FilterRules/degreesofseparation.py:126]: You need to make a test for no person here (None), in case the filter is in use with a different tree and root handle is not defined.  See https://gramps-project.org/bugs/view.php?id=12426
